src/python_testing/TC_MCORE_FS_1_4.py

```diff
@@ -0,0 +1,482 @@
+# when reading UniqueID from BasicInformation Cluster. Please specify the app
+# location with --string-arg th_server_no_uid_app_path:<path_to_app>
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs: run1
+# test-runner-run/run1/app: examples/fabric-admin/scripts/fabric-sync-app.py
+# test-runner-run/run1/app-args: --app-admin=${FABRIC_ADMIN_APP} --app-bridge=${FABRIC_BRIDGE_APP} --stdin-pipe=dut-fsa-stdin --discriminator=1234
+# test-runner-run/run1/factoryreset: True
+# test-runner-run/run1/script-args: --storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --string-arg th_fsa_admin_path:${FABRIC_ADMIN_APP} th_fsa_bridge_path:${FABRIC_BRIDGE_APP} th_server_no_uid_app_path:${LIGHTING_APP_NO_UNIQUE_ID} dut_fsa_stdin_pipe:dut-fsa-stdin
+# test-runner-run/run1/script-start-delay: 5
+# test-runner-run/run1/quiet: false
+# === END CI TEST ARGUMENTS ===
+
+import asyncio
+import logging
+import os
+import random
+import subprocess
+import sys
+import tempfile
+import threading
+
+import chip.clusters as Clusters
+from chip import ChipDeviceCtrl
+from chip.interaction_model import Status
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main, type_matches
+from mobly import asserts
+
+
+class Subprocess(threading.Thread):
+
+    def __init__(self, args: list = [], stdout_cb=None, tag="", **kw):
+        super().__init__(**kw)
+        self.tag = f"[{tag}] " if tag else ""
+        self.stdout_cb = stdout_cb
+        self.args = args
+
+    def forward_f(self, f_in, f_out):
+        while True:
+            line = f_in.readline()
+            if not line:
+                break
+            f_out.write(f"{self.tag}{line}")
+            f_out.flush()
+            if self.stdout_cb is not None:
+                self.stdout_cb(line)
+
+    def run(self):
+        logging.info("RUN: %s", " ".join(self.args))
+        self.p = subprocess.Popen(self.args, errors="ignore", stdin=subprocess.PIPE,
+                                  stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        # Forward stdout and stderr with a tag attached.
+        t1 = threading.Thread(target=self.forward_f, args=[self.p.stdout, sys.stdout])
+        t1.start()
+        t2 = threading.Thread(target=self.forward_f, args=[self.p.stderr, sys.stderr])
+        t2.start()
+        # Wait for the process to finish.
+        self.p.wait()
+        t1.join()
+        t2.join()
+
+    def stop(self):
+        self.p.terminate()
+        self.join()
+
+
+class FabricSyncApp:
+
+    APP_PATH = "examples/fabric-admin/scripts/fabric-sync-app.py"
+    FABRIC_ADMIN_PATH = "out/linux-x64-fabric-admin-rpc/fabric-admin"
+    FABRIC_BRIDGE_PATH = "out/linux-x64-fabric-bridge-rpc/fabric-bridge-app"
```

Comment:
This will need to be provided as command line arguments for this to work on TH. These paths do not exist on TH

Reply:
These values are only defaults. Everything can (and should be) provided via string args

Reply:
Removed default